From c5bf032a441db13a5b295f2d3414a3d99d9e1cc0 Mon Sep 17 00:00:00 2001
From: Christian Epple <christian.epple@aisec.fraunhofer.de>
Date: Fri, 5 Feb 2021 10:03:59 +0100
Subject: [PATCH 1/1] Fix includes for MUSL builds

Make sure unistd.h is included also when sc-hsm-embedded is configured to be used with --enable-ctapi
---
 src/pkcs11/p11generic.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/pkcs11/p11generic.h b/src/pkcs11/p11generic.h
index e87b0f0..3e3b71b 100644
--- a/src/pkcs11/p11generic.h
+++ b/src/pkcs11/p11generic.h
@@ -50,12 +50,14 @@
 #define _MAX_PATH FILENAME_MAX
 #endif
 
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
 #ifndef CTAPI
 #ifdef _WIN32
 #include <winscard.h>
 #define  MAX_READERNAME   128
-#else
-#include <unistd.h>
 #ifdef __APPLE__
 #include <PCSC/pcsclite.h>
 #include <PCSC/winscard.h>
-- 
2.20.1

